x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationUtils.java

```diff
@@ -113,6 +114,7 @@ public static void switchUserBasedOnActionOriginAndExecute(ThreadContext threadC
             case PERSISTENT_TASK_ORIGIN:
             case ROLLUP_ORIGIN:
             case INDEX_LIFECYCLE_ORIGIN:
+            case TASKS_ORIGIN:   // TODO use a more limited user for tasks
```

Comment:
I feel like we should add this user as part of this change

Reply:
I want to add the user, but I was thinking that adding the user would involve backwards compatibility for sending requests to nodes that don't have the user which would also involve adding backwards compatibility tests which felt to me like a bigger change than I wanted to bundle with this one. Do you buy that?

Reply:
I understand what you’re saying. If we are ok with this being broken as it is today in a mixed cluster then I don’t think BWC is a big concern. The user will get sent to a node that cannot resolve the role and it will fail authorization which is the same as what it does today. Switching to a new user will also technically break the API for users that had permissions to this index when they run in a mixed cluster and issue a request from the new node that hits a old node.